core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -476,7 +476,7 @@ private void updateMetrics(boolean success, Instant startTime,
    * be synced as well.
    *
    * @param path The path to sync
-   * @return true if this path was synced
+   * @return true if the system has valid cache for this path
    */
   private boolean processSyncPath(AlluxioURI path) {
```

Comment:
I don't think a boolean here is sufficient anymore. Now the states are at least:
1. Sync-ed successfully
2. Sync failed
3. Skipped because of the timestamp
4. Interrupted

I think switching to an enum will be a lot more expressive. We can start with something like:
```
enum SyncState {
  SKIPPED, SUCCESS, FAIL
}

// then
private SyncState processSyncPath()
```

Then in the InodeSyncStream, other than the existing `syncPathCount` and `failedSyncPathCount`, there should be one `skippedPathCount`. And we rely on that to determine whether the InodeSyncStream is a success or failure.

Honestly I don't really like how `InodeSyncStream.sync()` calls `notifySyncedPath()` only at the end. But that's out of the scope of this PR. I will discuss with @tcrain off line.